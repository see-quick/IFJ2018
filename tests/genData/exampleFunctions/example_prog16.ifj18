# Leap year

def get_year()
    print "Enter a year: \n"
    year = inputi

end

def modulo(divident, divider)
    res = divident - (divident / divider) * divider
end

year = get_year()
res = modulo(year, 4)
if(res == 0) then
    res = modulo(year, 100)
    if(res == 0) then
        # year is divisible by 400, hence the year is a leap year
        res = modulo(year, 400)
        if (res == 0) then
            print year, " is a leap year\n"
        else
            print year, " is not a leap year\n"
        end
    else
        print year, " is a leap year\n"
    end

else
     print year, " is not a leap year\n"
end
