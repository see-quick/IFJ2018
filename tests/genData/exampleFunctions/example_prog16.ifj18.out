1900 is a leap year

